repos/system_upgrade/common/actors/checkconsumedassets/libraries/check_consumed_assets.py

```diff
@@ -0,0 +1,162 @@
+import re
+from collections import defaultdict, namedtuple
+
+from leapp import reporting
+from leapp.libraries.common.config import get_consumed_data_stream_id
+from leapp.libraries.common.fetch import ASSET_PROVIDED_DATA_STREAMS_FIELD
+from leapp.libraries.stdlib import api
+from leapp.models import ConsumedDataAsset
+
+
+def compose_summary_for_incompatible_assets(assets, incompatibility_reason):
+    if not assets:
+        return []
+
+    summary_lines = ['The following assets are {reason}'.format(reason=incompatibility_reason)]
+    for asset in assets:
+        if asset.provided_data_streams is None:  # Assets with missing streams are placed only in .outdated bucket
+            details = (' - The asset {what_asset} is missing information about provided data streams '
+                       'in its metadata header')
+            details = details.format(what_asset=asset.filename)
+        else:
+            article, multiple_suffix = ('the ', '') if len(asset.provided_data_streams) == 1 else ('', 's')
+            details = ' - The asset {what_asset} provides data {article}stream{mult_suffix} {provided_streams}'
+            details = details.format(what_asset=asset.filename,
+                                     provided_streams=', '.join(asset.provided_data_streams),
+                                     article=article, mult_suffix=multiple_suffix)
+        summary_lines.append(details)
+    return summary_lines
+
+
+def make_report_entries_with_unique_urls(docs_url_to_title_map):
+    report_urls = []
+    # Add every unique asset URL into the report
+    urls_with_multiple_titles = []
+    for url, titles in docs_url_to_title_map.items():
+        if len(titles) > 1:
+            urls_with_multiple_titles.append(url)
+
+        report_entry = reporting.ExternalLink(title=titles[0], url=url)
+        report_urls.append(report_entry)
+
+    if urls_with_multiple_titles:
+        msg = 'Docs URLs {urls} are used with inconsistent URL titles, picking one.'
+        api.current_logger().warning(msg.format(', '.join(urls_with_multiple_titles)))
```

Comment:
Missing `urls=','.join(urls_with_multiple_titles)` in call to format.

Reply:
fixed